### DISCUSSION TITLE
Built-in, URP or HDRP

### DISCUSSION THREAD

**Discussion by Sythelux** [V-Sekai/unidot_importer#47] [Q&A] [answered]
On some asset stores I have the ability to choose between built-in Unity pipeline or Universal Render Pipeline or High Definition Render Pipeline.

Which would work best with this toolkit at the moment?

I do suspect, that you try to achieve compatibility with all, but I would still like to know if there is a preferred way.

**Comment by lyuma** [comment, marked as answer]
Currently Unidot is designed for built-in render pipeline assets, in particular those using Standard shader or with compatible naming (it is common for shaders to use compatible names since they often specify a fallback shader such as Standard or Diffuse).

The reason is because scriptable render pipeline assets often do not use a "Standard" shader but instead tend to define a custom shader using shader graph, which is not currently supported, and the texture inputs are often random numbers, so Unidot has to guess which texture is which. I find the guess of albedo texture to be correct most of the time in my assets, but I would not be surprised if it fails more often than not.

I do hope to someday have some support at least for simple shader graphs, but until then, we are limited to guessing texture mappings in the URP/HDRP case.

**Reply by lyuma** [reply]
As for HDRP-compatible lights, Godot should support using nits for lighting, but your Godot project setting must be configured to use physical lighting units. See the documentation: https://docs.godotengine.org/en/stable/tutorials/3d/physical_light_and_camera_units.html

**Reply by Sythelux** [reply]
Thank you. I thought it might be that, but I wanted to make sure as a possible scenario could've been "use URP we had a breakthrough and everything is compatible with URP now".

but I'm fine with "use built-in that works best".


as for the physical based lighting. Do you think it would be a good idea to have an unidot wizard that enables such things?
